src/ui/common/wireRouting.cpp

```
@@ -0,0 +1,143 @@
+      Candidate{QPointF(point.x(), normalized.bottom()),
+                (point.y() - normalized.bottom()) * (point.y() - normalized.bottom())},
+  };
+
+  const auto closest =
+      std::ranges::min_element(candidates, {}, &Candidate::distanceSquared);
+  return closest->point;
+}
+
+std::vector<QPointF> removeConsecutiveDuplicates(std::vector<QPointF> points)
+{
+  points.erase(std::ranges::unique(points).begin(), points.end());
```

Cppcheck (2.20.0)

[warning] Iterators to containers from different expressions 'std

(mismatchingContainerExpression)